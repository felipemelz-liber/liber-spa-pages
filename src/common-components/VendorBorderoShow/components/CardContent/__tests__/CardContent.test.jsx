import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { CardContent } from 'liber-components';

describe('CardContent component tests', () => {
  configure({ adapter: new Adapter() });

  const baseSummary = {
    buyer: { name: 'Mock buyer', cnpj: '06.194.174/0001-19' },
    paidValue: 'R$ 1234',
    invoiceCount: '20',
    paidInvoiceCount: '18',
    requestedAt: '12/12/2020',
    bankAccount: {
      branch: '001',
      branchCd: '1',
      account: '12345',
      accountCd: '1',
      bankName: 'Banco Mockado',
      id: 1,
      bankCode: '001',
    },
  };

  const cases = [
    {
      summary: {
        ...baseSummary,
        signatures: { required: 2 },
        operators: [
          {
            id: 1,
            name: 'User 1',
            userHasSigned: true,
          },
          {
            id: 1,
            name: 'User 2',
            userHasSigned: false,
          },
        ],
        signatureCertificateUrl: null,
        invoiceTransferAgreementUrl: null,
      },
    },
    {
      summary: {
        ...baseSummary,
        signatures: { required: 2 },
        operators: [
          {
            id: 1,
            name: 'User 1',
            userHasSigned: true,
          },
          {
            id: 1,
            name: 'User 2',
            userHasSigned: true,
          },
        ],
        signatureCertificateUrl: null,
        invoiceTransferAgreementUrl: null,
      },
    },
    {
      summary: {
        ...baseSummary,
        signatures: { required: 1 },
        operators: [
          {
            id: 1,
            name: 'User 1',
            userHasSigned: true,
          },
          {
            id: 1,
            name: 'User 2',
            userHasSigned: false,
          },
        ],
        signatureCertificateUrl: 'mockUrl1',
        invoiceTransferAgreementUrl: 'mockUrl2',
      },
    },
    {
      summary: {
        ...baseSummary,
        signatures: { required: 1 },
        operators: [
          {
            id: 1,
            name: 'User 1',
            userHasSigned: false,
          },
        ],
        signatureCertificateUrl: null,
        invoiceTransferAgreementUrl: 'mockUrl2',
      },
    },
  ];

  it('should render CardContent correctly without props', () => {
    const wrapper = shallow(<CardContent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  cases.forEach((props, index) => {
    it(`should render CardContent correctly for case ${index} with props`, () => {
      const wrapper = shallow(<CardContent {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
