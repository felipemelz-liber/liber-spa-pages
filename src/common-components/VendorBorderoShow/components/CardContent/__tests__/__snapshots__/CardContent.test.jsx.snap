// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent component tests should render CardContent correctly for case 0 with props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Cards__CardContent-sc-1kn0psw-2",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Cards__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Cards__CardContent-sc-1kn0psw-2",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  summary={
    Object {
      "bankAccount": Object {
        "account": "12345",
        "accountCd": "1",
        "bankCode": "001",
        "bankName": "Banco Mockado",
        "branch": "001",
        "branchCd": "1",
        "id": 1,
      },
      "buyer": Object {
        "cnpj": "06.194.174/0001-19",
        "name": "Mock buyer",
      },
      "invoiceCount": "20",
      "invoiceTransferAgreementUrl": null,
      "operators": Array [
        Object {
          "id": 1,
          "name": "User 1",
          "userHasSigned": true,
        },
        Object {
          "id": 1,
          "name": "User 2",
          "userHasSigned": false,
        },
      ],
      "paidInvoiceCount": "18",
      "paidValue": "R$ 1234",
      "requestedAt": "12/12/2020",
      "signatureCertificateUrl": null,
      "signatures": Object {
        "required": 2,
      },
    }
  }
/>
`;

exports[`CardContent component tests should render CardContent correctly for case 1 with props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Cards__CardContent-sc-1kn0psw-2",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Cards__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Cards__CardContent-sc-1kn0psw-2",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  summary={
    Object {
      "bankAccount": Object {
        "account": "12345",
        "accountCd": "1",
        "bankCode": "001",
        "bankName": "Banco Mockado",
        "branch": "001",
        "branchCd": "1",
        "id": 1,
      },
      "buyer": Object {
        "cnpj": "06.194.174/0001-19",
        "name": "Mock buyer",
      },
      "invoiceCount": "20",
      "invoiceTransferAgreementUrl": null,
      "operators": Array [
        Object {
          "id": 1,
          "name": "User 1",
          "userHasSigned": true,
        },
        Object {
          "id": 1,
          "name": "User 2",
          "userHasSigned": true,
        },
      ],
      "paidInvoiceCount": "18",
      "paidValue": "R$ 1234",
      "requestedAt": "12/12/2020",
      "signatureCertificateUrl": null,
      "signatures": Object {
        "required": 2,
      },
    }
  }
/>
`;

exports[`CardContent component tests should render CardContent correctly for case 2 with props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Cards__CardContent-sc-1kn0psw-2",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Cards__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Cards__CardContent-sc-1kn0psw-2",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  summary={
    Object {
      "bankAccount": Object {
        "account": "12345",
        "accountCd": "1",
        "bankCode": "001",
        "bankName": "Banco Mockado",
        "branch": "001",
        "branchCd": "1",
        "id": 1,
      },
      "buyer": Object {
        "cnpj": "06.194.174/0001-19",
        "name": "Mock buyer",
      },
      "invoiceCount": "20",
      "invoiceTransferAgreementUrl": "mockUrl2",
      "operators": Array [
        Object {
          "id": 1,
          "name": "User 1",
          "userHasSigned": true,
        },
        Object {
          "id": 1,
          "name": "User 2",
          "userHasSigned": false,
        },
      ],
      "paidInvoiceCount": "18",
      "paidValue": "R$ 1234",
      "requestedAt": "12/12/2020",
      "signatureCertificateUrl": "mockUrl1",
      "signatures": Object {
        "required": 1,
      },
    }
  }
/>
`;

exports[`CardContent component tests should render CardContent correctly for case 3 with props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Cards__CardContent-sc-1kn0psw-2",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Cards__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Cards__CardContent-sc-1kn0psw-2",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  summary={
    Object {
      "bankAccount": Object {
        "account": "12345",
        "accountCd": "1",
        "bankCode": "001",
        "bankName": "Banco Mockado",
        "branch": "001",
        "branchCd": "1",
        "id": 1,
      },
      "buyer": Object {
        "cnpj": "06.194.174/0001-19",
        "name": "Mock buyer",
      },
      "invoiceCount": "20",
      "invoiceTransferAgreementUrl": "mockUrl2",
      "operators": Array [
        Object {
          "id": 1,
          "name": "User 1",
          "userHasSigned": false,
        },
      ],
      "paidInvoiceCount": "18",
      "paidValue": "R$ 1234",
      "requestedAt": "12/12/2020",
      "signatureCertificateUrl": null,
      "signatures": Object {
        "required": 1,
      },
    }
  }
/>
`;

exports[`CardContent component tests should render CardContent correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Cards__CardContent-sc-1kn0psw-2",
        "isStatic": false,
        "rules": Array [],
      },
      "displayName": "Cards__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Cards__CardContent-sc-1kn0psw-2",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
